### PR TITLE
add branchName and commitId to s2iconfig

### PR DESCRIPTION
**Pull request by soulseen**
Signed-off-by: soulseen <sunzhu@yunify.com>

### CODE REVIEW

File: pkg/controller/s2irun/ksbuilder_jobs.go [hunk 1]
```diff
@@ -278,37 +278,20 @@ func (r *ReconcileS2iRun) setGitSecret(instance *devopsv1alpha1.S2iRun, config *
...
+		}
+		password, ok := secret.Data[corev1.BasicAuthPasswordKey]
+		if !ok {
+			return fmt.Errorf("could not get password in secret %s", secret.Name)
+		}
+		sourceUrl, err := url.Parse(config.SourceURL)
+		if err != nil {
+			return err
 		}
+		config.SourceURL = fmt.Sprintf("%s://%s:%s@%s%s", sourceUrl.Scheme, url.QueryEscape(string(username)), url.QueryEscape(string(password)), sourceUrl.Host, sourceUrl.RequestURI())
```
Comment (runzexia) [inline pkg/controller/s2irun/ksbuilder_jobs.go]: Should consider the case where no branch and commit.

Comment (soulseen) [reply pkg/controller/s2irun/ksbuilder_jobs.go]: Here just formatted SourceURL，which does not have relationship with branch and commit. Branch and commit are set in S2iBuilder.spec，it  also works well even if they are not set.

**Comment by soulseen** [comment]
ok.

**Comment by magicsong** [comment]
/hold cancel
/approve
cc @runzexia

**Comment by runzexia** [comment]
I think we need to improve s2irun and specify commit every time I trigger s2i run. So that we can combine with webhook

**Comment by soulseen** [comment]
https://github.com/kubesphere/s2irun/pull/1/files#diff-b09a606faad762d657b55d228383da05R26

**Comment by soulseen** [comment]
@runzexia

**Pull request merged**